typing/env.ml

```diff
@@ -2921,32 +2935,55 @@ let lock_mode ~errors ~loc env id vmode locks =
     (fun (vmode, reason) lock ->
       match lock with
       | Region_lock -> (Mode.Value.local_to_regional vmode, reason)
-      | Locality_lock {mode; escaping_context} -> begin
+      | Escape_lock escaping_context -> begin
           match
             Mode.Regionality.submode
               (Mode.Value.locality vmode)
-              (Mode.Regionality.of_locality mode)
+              (Mode.Regionality.global)
           with
           | Ok () -> (vmode, reason)
           | Error _ ->
               may_lookup_error errors loc env
-                (Local_value_used_in_closure (id, escaping_context))
+                (Local_value_escaping (id, escaping_context))
+        end
+      | Share_lock shared_context -> begin
+          ( match
```

Comment:
Weird space between `match` and `(`. Also, you don't need both the `begin ... end` and the parentheses.